lib/LedgerSMB/I18N.pm

```diff
@@ -135,6 +137,29 @@ sub get_country_list {
     ];
 }
 
+=item get_language_list($language)
+
+Get a language localized list to allow user selection
+
+=cut
+
+sub get_language_list {
+    my ($self,$language) = @_;
+    my $locale = Locales->new($language);
+    my @rows = $self->call_dbmethod(funcname => 'person__list_languages');
+    my @languages;
+    for (@rows) {
+        my $text = ucfirst($locale->get_language_from_code(lc($_->{code})) // $_->{description});
+        utf8::decode($text);
```

Comment:
shouldn't we decode the text *before* doing the `ucfirst` transformation?

Reply:
(and maybe we should make sure not to decode `$_->{description}` ?

Reply:
or instead, maybe we need to use this hint: https://metacpan.org/pod/Locales#To-byte-or-not-to-byte,-that-is-the-question.-Ok,-I%E2%80%99ll-byte-%E2%80%A6

Reply:
just one quick question (finding this only now; would have asked earlier if I had found it before): is https://metacpan.org/pod/Locale::CLDR#all_languages() helpful in reducing the extra dependency?

Reply:
Much better with unicode. Fixed.
Using all_languages would bring back tons of languages, including sign, with only english descriptions. Thus we cannot display them in user language.